From c72f1c022b3a4b100063e96b744392a815ed26ce Mon Sep 17 00:00:00 2001
From: Daniel Ammann <daniel.ammann@bytesatwork.ch>
Date: Mon, 2 Mar 2020 15:11:13 +0100
Subject: [PATCH] byteengine_m2: Add support for 2020.01

Remove support for byteENGINE AM335x with NAND.

This commit includes fixes for the following changes:
- f3998fdc4d env: Rename environment.h to env_internal.h
- 88718be300 mtd: rename CONFIG_NAND -> CONFIG_MTD_RAW_NAND
- 6ae3900a86 mtd: nand: Rename nand.h into rawnand.h
- 0c3a9ed409 spl: Kconfig: Replace CONFIG_SPL_FAT_SUPPORT with CONFIG_SPL_FS_FAT
---
 arch/arm/dts/Makefile                         |   1 +
 arch/arm/dts/am335x-bytedevkit-nand.dts       |  22 ----
 ...edevkit-emmc.dts => am335x-bytedevkit.dts} |   0
 .../bytesatwork/byteengine_am335x/MAINTAINERS |   7 -
 board/bytesatwork/byteengine_am335x/board.c   |   5 +-
 board/bytesatwork/byteengine_am335x/mux.c     |  21 ---
 common/command.c                              |   3 +-
 ..._defconfig => am335x_bytedevkit_defconfig} |  10 +-
 configs/am335x_bytedevkit_nand_defconfig      |  52 --------
 include/configs/byteengine_am335x.h           | 121 +-----------------
 10 files changed, 12 insertions(+), 230 deletions(-)
 delete mode 100644 arch/arm/dts/am335x-bytedevkit-nand.dts
 rename arch/arm/dts/{am335x-bytedevkit-emmc.dts => am335x-bytedevkit.dts} (100%)
 delete mode 100644 board/bytesatwork/byteengine_am335x/MAINTAINERS
 rename configs/{am335x_bytedevkit_emmc_defconfig => am335x_bytedevkit_defconfig} (88%)
 delete mode 100644 configs/am335x_bytedevkit_nand_defconfig

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 79b60ee0cb..e0d21fe226 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -304,6 +304,7 @@ dtb-$(CONFIG_AM33XX) += \
 	am335x-brppt1-spi.dtb \
 	am335x-brxre1.dtb \
 	am335x-brsmarc1.dtb \
+	am335x-bytedevkit.dtb \
 	am335x-draco.dtb \
 	am335x-evm.dtb \
 	am335x-evm-mini.dtb \
diff --git a/arch/arm/dts/am335x-bytedevkit-nand.dts b/arch/arm/dts/am335x-bytedevkit-nand.dts
deleted file mode 100644
index 4516117eb4..0000000000
--- a/arch/arm/dts/am335x-bytedevkit-nand.dts
+++ /dev/null
@@ -1,22 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0
-/*
- * Copyright (C) 2018 bytes at work AG
- *
- * Based on am335x-evm.dts
- *
- * Copyright (C) 2012 Texas Instruments Incorporated - http://www.ti.com/
- */
-
-/dts-v1/;
-
-#include "am335x-byteengine.dtsi"
-#include "am335x-bytedevkit.dtsi"
-
-/ {
-	model = "bytesatwork byteDEVKIT with byteENGINE AM335x NAND";
-	compatible = "bytesatwork,byteengine-am335x", "ti,am33xx";
-};
-
-&gpmc {
-	status = "okay";
-};
diff --git a/arch/arm/dts/am335x-bytedevkit-emmc.dts b/arch/arm/dts/am335x-bytedevkit.dts
similarity index 100%
rename from arch/arm/dts/am335x-bytedevkit-emmc.dts
rename to arch/arm/dts/am335x-bytedevkit.dts
diff --git a/board/bytesatwork/byteengine_am335x/MAINTAINERS b/board/bytesatwork/byteengine_am335x/MAINTAINERS
deleted file mode 100644
index 13ba4d13f4..0000000000
--- a/board/bytesatwork/byteengine_am335x/MAINTAINERS
+++ /dev/null
@@ -1,7 +0,0 @@
-BYTEENGINE_AM335X
-M:	Markus Kappeler <markus.kappeler@bytesatwork.ch>
-S:	Maintained
-F:	board/bytesatwork/byteengine_am335x/
-F:	include/configs/byteengine_am335x.h
-F:	configs/am335x_bytedevkit_emmc_defconfig
-F:	configs/am335x_bytedevkit_nand_defconfig
diff --git a/board/bytesatwork/byteengine_am335x/board.c b/board/bytesatwork/byteengine_am335x/board.c
index 397159d32b..79043090ed 100644
--- a/board/bytesatwork/byteengine_am335x/board.c
+++ b/board/bytesatwork/byteengine_am335x/board.c
@@ -17,7 +17,7 @@
 #include <asm/io.h>
 #include <cpsw.h>
 #include <power/tps65910.h>
-#include <environment.h>
+#include <env_internal.h>
 #include <errno.h>
 #include <miiphy.h>
 #include <spl.h>
@@ -87,9 +87,6 @@ int board_init(void)
 #endif
 
 	gd->bd->bi_boot_params = CONFIG_SYS_SDRAM_BASE + 0x100;
-#ifdef CONFIG_NAND
-	gpmc_init();
-#endif
 
 	return 0;
 }
diff --git a/board/bytesatwork/byteengine_am335x/mux.c b/board/bytesatwork/byteengine_am335x/mux.c
index a1aa777fc8..ae098b67e1 100644
--- a/board/bytesatwork/byteengine_am335x/mux.c
+++ b/board/bytesatwork/byteengine_am335x/mux.c
@@ -45,25 +45,6 @@ static struct module_pin_mux mmc1_pin_mux[] = {
 	{OFFSET(gpmc_csn2), (MODE(2) | RXACTIVE | PULLUP_EN)},	/* MMC1_CMD */
 	{-1},
 };
-#else
-static struct module_pin_mux nand_pin_mux[] = {
-	{OFFSET(gpmc_ad0), (MODE(0) | RXACTIVE | PULLUP_EN)},	/* GPMC_AD0 */
-	{OFFSET(gpmc_ad1), (MODE(0) | RXACTIVE | PULLUP_EN)},	/* GPMC_AD1 */
-	{OFFSET(gpmc_ad2), (MODE(0) | RXACTIVE | PULLUP_EN)},	/* GPMC_AD2 */
-	{OFFSET(gpmc_ad3), (MODE(0) | RXACTIVE | PULLUP_EN)},	/* GPMC_AD3 */
-	{OFFSET(gpmc_ad4), (MODE(0) | RXACTIVE | PULLUP_EN)},	/* GPMC_AD4 */
-	{OFFSET(gpmc_ad5), (MODE(0) | RXACTIVE | PULLUP_EN)},	/* GPMC_AD5 */
-	{OFFSET(gpmc_ad6), (MODE(0) | RXACTIVE | PULLUP_EN)},	/* GPMC_AD6 */
-	{OFFSET(gpmc_ad7), (MODE(0) | RXACTIVE | PULLUP_EN)},	/* GPMC_AD7 */
-	{OFFSET(gpmc_wpn), (MODE(7) | RXACTIVE | PULLUP_EN)},	/* GPMC_WPN */
-	{OFFSET(gpmc_csn0), (MODE(0))},				/* GPMC_CSN0 */
-	{OFFSET(gpmc_advn_ale), (MODE(0))},			/* GPMC_ADVN_ALE */
-	{OFFSET(gpmc_be0n_cle), (MODE(0))},			/* GPMC_BEN0_CLE */
-	{OFFSET(gpmc_wen), (MODE(0))},				/* GPMC_WEN */
-	{OFFSET(gpmc_oen_ren), (MODE(0))},			/* GPMC_OEN_REN */
-	{OFFSET(gpmc_wait0), (MODE(0) | RXACTIVE | PULLUP_EN)},	/* GPMC_WAIT0 */
-	{-1},
-};
 #endif
 
 static struct module_pin_mux i2c0_pin_mux[] = {
@@ -138,7 +119,5 @@ void enable_board_pin_mux(void)
 #ifdef CONFIG_SPI_FLASH
 	configure_module_pin_mux(spi0_pin_mux);
 	configure_module_pin_mux(mmc1_pin_mux);
-#else
-	configure_module_pin_mux(nand_pin_mux);
 #endif
 }
diff --git a/common/command.c b/common/command.c
index ceca992510..16b33ca00c 100644
--- a/common/command.c
+++ b/common/command.c
@@ -587,11 +587,12 @@ enum command_ret_t cmd_process(int flag, int argc, char * const argv[],
 
 #if defined(CONFIG_SYS_XTRACE)
 	char *xtrace;
+	int i;
 
 	xtrace = env_get("xtrace");
 	if (xtrace) {
 		puts("+");
-		for (int i = 0; i < argc; i++) {
+		for (i = 0; i < argc; i++) {
 			puts(" ");
 			puts(argv[i]);
 		}
diff --git a/configs/am335x_bytedevkit_emmc_defconfig b/configs/am335x_bytedevkit_defconfig
similarity index 88%
rename from configs/am335x_bytedevkit_emmc_defconfig
rename to configs/am335x_bytedevkit_defconfig
index b6089c76f9..01e73833b9 100644
--- a/configs/am335x_bytedevkit_emmc_defconfig
+++ b/configs/am335x_bytedevkit_defconfig
@@ -6,8 +6,11 @@ CONFIG_AM33XX=y
 CONFIG_TARGET_BYTEENGINE_AM335X=y
 CONFIG_SPL_MMC_SUPPORT=y
 CONFIG_SPL_SERIAL_SUPPORT=y
+CONFIG_ENV_SIZE=0x10000
+CONFIG_ENV_OFFSET=0x210000
 CONFIG_SPL=y
-CONFIG_SPL_FAT_SUPPORT=y
+CONFIG_ENV_SECT_SIZE=0x10000
+CONFIG_SPL_FS_FAT=y
 CONFIG_SPL_LIBDISK_SUPPORT=y
 CONFIG_SPL_SPI_FLASH_SUPPORT=y
 CONFIG_SPL_SPI_SUPPORT=y
@@ -20,7 +23,6 @@ CONFIG_SPL_WATCHDOG_SUPPORT=y
 CONFIG_HUSH_PARSER=y
 CONFIG_CMD_CPUINFO=y
 CONFIG_CMD_MEMINFO=y
-# CONFIG_CMD_FLASH is not set
 CONFIG_CMD_GPIO=y
 CONFIG_CMD_I2C=y
 CONFIG_CMD_MMC=y
@@ -33,15 +35,15 @@ CONFIG_CMD_PXE=y
 CONFIG_CMD_EXT4=y
 CONFIG_CMD_FAT=y
 CONFIG_CMD_FS_GENERIC=y
-CONFIG_CMD_MTDPARTS=y
 CONFIG_OF_CONTROL=y
-CONFIG_DEFAULT_DEVICE_TREE="am335x-bytedevkit-emmc"
+CONFIG_DEFAULT_DEVICE_TREE="am335x-bytedevkit"
 CONFIG_ENV_IS_IN_SPI_FLASH=y
 CONFIG_DM_I2C=y
 CONFIG_MISC=y
 CONFIG_DM_MMC=y
 CONFIG_MMC_OMAP_HS=y
 CONFIG_SPI_FLASH=y
+CONFIG_SF_DEFAULT_SPEED=24000000
 CONFIG_SPI_FLASH_SFDP_SUPPORT=y
 CONFIG_SPI_FLASH_ATMEL=y
 CONFIG_SPI_FLASH_EON=y
diff --git a/configs/am335x_bytedevkit_nand_defconfig b/configs/am335x_bytedevkit_nand_defconfig
deleted file mode 100644
index 15c5bd1fe3..0000000000
--- a/configs/am335x_bytedevkit_nand_defconfig
+++ /dev/null
@@ -1,52 +0,0 @@
-CONFIG_ARM=y
-CONFIG_ARCH_OMAP2PLUS=y
-CONFIG_SPL_LIBCOMMON_SUPPORT=y
-CONFIG_SPL_LIBGENERIC_SUPPORT=y
-CONFIG_AM33XX=y
-CONFIG_TARGET_BYTEENGINE_AM335X=y
-CONFIG_SPL_MMC_SUPPORT=y
-CONFIG_SPL_SERIAL_SUPPORT=y
-CONFIG_SPL=y
-CONFIG_SPL_FAT_SUPPORT=y
-CONFIG_SPL_LIBDISK_SUPPORT=y
-CONFIG_SYS_CONSOLE_INFO_QUIET=y
-CONFIG_SPL_I2C_SUPPORT=y
-CONFIG_SPL_POWER_SUPPORT=y
-CONFIG_SPL_WATCHDOG_SUPPORT=y
-CONFIG_HUSH_PARSER=y
-CONFIG_CMD_CPUINFO=y
-CONFIG_CMD_MEMINFO=y
-# CONFIG_CMD_FLASH is not set
-CONFIG_CMD_GPIO=y
-CONFIG_CMD_I2C=y
-CONFIG_CMD_MMC=y
-CONFIG_CMD_NAND=y
-# CONFIG_CMD_SETEXPR is not set
-CONFIG_CMD_DHCP=y
-CONFIG_CMD_MII=y
-CONFIG_CMD_PING=y
-CONFIG_CMD_PXE=y
-CONFIG_CMD_EXT4=y
-CONFIG_CMD_FAT=y
-CONFIG_CMD_FS_GENERIC=y
-CONFIG_CMD_MTDPARTS=y
-CONFIG_MTDIDS_DEFAULT="nand0=8000000.nand"
-CONFIG_MTDPARTS_DEFAULT="mtdparts=8000000.nand:128k(NAND.SPL),128k(NAND.SPL.backup1),128k(NAND.SPL.backup2),128k(NAND.SPL.backup3),256k(NAND.u-boot-spl-os),1m(NAND.u-boot),128k(NAND.u-boot-env),128k(NAND.u-boot-env.backup1),8m(NAND.kernel),-(NAND.file-system)"
-CONFIG_OF_CONTROL=y
-CONFIG_DEFAULT_DEVICE_TREE="am335x-bytedevkit-nand"
-CONFIG_ENV_IS_IN_NAND=y
-CONFIG_DM_I2C=y
-CONFIG_MISC=y
-CONFIG_DM_MMC=y
-CONFIG_MMC_OMAP_HS=y
-CONFIG_NAND=y
-CONFIG_PHY_TI=y
-CONFIG_DM_ETH=y
-CONFIG_MII=y
-CONFIG_DRIVER_TI_CPSW=y
-CONFIG_SPI=y
-CONFIG_DM_SPI=y
-CONFIG_OMAP3_SPI=y
-CONFIG_TIMER=y
-CONFIG_OMAP_TIMER=y
-# CONFIG_EFI_LOADER is not set
diff --git a/include/configs/byteengine_am335x.h b/include/configs/byteengine_am335x.h
index d75406c67b..9198b870f2 100644
--- a/include/configs/byteengine_am335x.h
+++ b/include/configs/byteengine_am335x.h
@@ -51,17 +51,7 @@
 #undef CONFIG_DM_USB
 #endif
 
-#ifdef CONFIG_NAND
-#define BOOT_ENV_SETTINGS	NAND_BOOT_ENV_SETTINGS
-#define CONFIG_BOOTCOMMAND \
-	"if mmc rescan; then " \
-		"echo SD/MMC found on device ${mmc_dev}; " \
-		"run nand_update; " \
-		"run mmc_boot; " \
-	"else " \
-		"run nand_boot; " \
-	"fi; "
-#else
+#ifdef CONFIG_SPI_FLASH
 #define BOOT_ENV_SETTINGS	MMC_BOOT_ENV_SETTINGS
 #define CONFIG_BOOTCOMMAND \
 	"if mmc rescan; then " \
@@ -102,110 +92,6 @@
 
 #endif
 
-#ifdef CONFIG_NAND
-
-/* NAND: device related configs */
-#define CONFIG_SYS_NAND_5_ADDR_CYCLE
-#define CONFIG_SYS_NAND_PAGE_COUNT	(CONFIG_SYS_NAND_BLOCK_SIZE / \
-					 CONFIG_SYS_NAND_PAGE_SIZE)
-#define CONFIG_SYS_NAND_PAGE_SIZE	2048
-#define CONFIG_SYS_NAND_OOBSIZE		64
-#define CONFIG_SYS_NAND_BLOCK_SIZE	SZ_128K
-/* NAND: driver related configs */
-#define CONFIG_SYS_NAND_BAD_BLOCK_POS	NAND_LARGE_BADBLOCK_POS
-#define CONFIG_SYS_NAND_ECCPOS		{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12}
-
-#define CONFIG_SYS_NAND_ECCSIZE		512
-#define CONFIG_SYS_NAND_ECCBYTES	3
-#define CONFIG_SYS_NAND_ONFI_DETECTION
-#define CONFIG_NAND_OMAP_ECCSCHEME	OMAP_ECC_HAM1_CODE_HW
-#define CONFIG_SYS_NAND_U_BOOT_OFFS	0x00080000
-/* NAND: SPL related configs */
-
-#define MTDPARTS_DEFAULT		"mtdparts=omap2-nand.0:128k(SPL)," \
-					"128k(SPL.backup1)," \
-					"128k(SPL.backup2)," \
-					"128k(SPL.backup3),1920k(u-boot)," \
-					"128k(u-boot-env),512k(devtree),5m(kernel),-(rootfs)"
-
-#ifdef CONFIG_ENV_IS_IN_NAND
-#define CONFIG_ENV_OFFSET		0x260000
-#define CONFIG_ENV_SIZE			SZ_128K
-#define CONFIG_SYS_ENV_SECT_SIZE	CONFIG_SYS_NAND_BLOCK_SIZE
-#else
-#define CONFIG_SYS_MMC_ENV_DEV		0
-#define CONFIG_ENV_OFFSET		SZ_128K
-#define CONFIG_ENV_SIZE			SZ_8K
-#endif
-
-#define NAND_BOOT_ENV_SETTINGS \
-	"mtd_parts=" MTDPARTS_DEFAULT "\0" \
-	"nand_root=/dev/mtdblock8\0" \
-	"nand_root_fs_type=jffs2\0" \
-	"nand_spl_addr=   0x00000000\0" \
-	"nand_spl_size=   0x00080000\0" \
-	"nand_u-boot_addr=0x00080000\0" \
-	"nand_u-boot_size=0x001e0000\0" \
-	"nand_param_addr= 0x00260000\0" \
-	"nand_param_size= 0x00020000\0" \
-	"nand_dtb_addr=   0x00280000\0" \
-	"nand_dtb_size=   0x00080000\0" \
-	"nand_kernel_addr=0x00300000\0" \
-	"nand_kernel_size=0x00500000\0" \
-	"nand_rootfs_addr=0x00800000\0" \
-	"nand_rootfs_size=0x0f800000\0" \
-	"updatepath=/batw-sw\0" \
-	"nand_args=setenv bootargs " \
-		"console=${console} " \
-		"vt.global_cursor_default=0 " \
-		"${mtd_parts} " \
-		"root=${nand_root} " \
-		"rootfstype=${nand_root_fs_type} " \
-		"rootwait " \
-		"consoleblank=0 " \
-		"; " \
-	"\0" \
-	"nand_boot=echo Booting from nand ...; " \
-		"run nand_args; " \
-		"nand read ${loadaddr} ${nand_kernel_addr} ${nand_kernel_size} || exit;" \
-		"nand read ${dtbaddr} ${nand_dtb_addr} ${nand_dtb_size} || exit; " \
-		"bootm ${loadaddr} - ${dtbaddr}; " \
-	"\0" \
-	"nand_update=echo Updating nand from mmc...; " \
-		"if size mmc 0 ${updatepath}; then " \
-			"if load mmc 0 ${loadaddr} ${updatepath}/MLO; then " \
-				"echo Found new SPL, flash it.; "\
-				"nand erase ${nand_spl_addr} ${nand_spl_size}; " \
-				"nand write ${loadaddr} ${nand_spl_addr} ${filesize}; " \
-			"fi; " \
-			"if load mmc 0 ${loadaddr} ${updatepath}/u-boot.img; then " \
-				"echo Found new U-Boot, flash it.; "\
-				"nand erase ${nand_u-boot_addr} ${nand_u-boot_size}; " \
-				"nand write ${loadaddr} ${nand_u-boot_addr} ${filesize}; " \
-			"fi; " \
-			"if load mmc 0 ${loadaddr} ${updatepath}/devtree.dtb; then " \
-				"echo Found new device tree, flash it.; "\
-				"nand erase ${nand_dtb_addr} ${nand_dtb_size}; " \
-				"nand write ${loadaddr} ${nand_dtb_addr} ${filesize}; " \
-			"fi; " \
-			"if load mmc 0 ${loadaddr} ${updatepath}/uImage; then " \
-				"echo Found new uImage, flash it.; "\
-				"nand erase ${nand_kernel_addr} ${nand_kernel_size}; " \
-				"nand write ${loadaddr} ${nand_kernel_addr} ${filesize}; " \
-			"fi; " \
-			"if load mmc 0 ${loadaddr} ${updatepath}/rootfs.jffs2; then " \
-				"echo Found new rootfs, flash it.; "\
-				"nand erase ${nand_rootfs_addr} ${nand_rootfs_size}; " \
-				"nand write ${loadaddr} ${nand_rootfs_addr} ${filesize}; " \
-			"fi; " \
-			"echo Flashing done. Remove SD card before power cycling.; " \
-		"else " \
-			"echo Update path ${updatepath} not found.; " \
-		"fi; " \
-	"\0"
-
-#endif
-
 #ifdef CONFIG_SPI_FLASH
 
 #ifndef CONFIG_SPL_BUILD
@@ -213,7 +99,7 @@
 #endif
 
 /* SPI Nor Flash */
-#define CONFIG_SF_DEFAULT_SPEED		24000000
+
 /*
  * Default to using SPI for environment for M25PE40.
  * 0x000000 - 0x010000 : SPL (64KiB)
@@ -226,10 +112,7 @@
 #define CONFIG_SYS_SPI_U_BOOT_OFFS	0x10000
 
 /* M25PE40: 64 KiB env size */
-#define CONFIG_ENV_SIZE			SZ_64K
 #define CONFIG_ENV_SPI_MAX_HZ		CONFIG_SF_DEFAULT_SPEED
-#define CONFIG_ENV_SECT_SIZE		SZ_64K
-#define CONFIG_ENV_OFFSET		(2112 * 1024)
 #define MTDIDS_DEFAULT			"nor0=m25pe40-flash.0"
 #define MTDPARTS_DEFAULT		"mtdparts=m25pe40-flash.0:64k(SPL)," \
 					"2048k(u-boot),64k(u-boot-env1)"
-- 
2.20.1

